### ISSUE TITLE
Lost of atoti extension

### ISSUE THREAD

**Issue by HuifangYeo** [atoti/atoti#211]
## Steps to reproduce
Not sure what triggered but occasionally I lose the extension and had no choice but to refresh the page for the extension to come back.

![image](https://user-images.githubusercontent.com/35210495/104551091-2a8b5680-5671-11eb-9572-b2d432516342.png)

## Environment

- atoti: 0.5.1
- Operating System: win32

**Comment by tibdex** [comment]
Hi Hui Fang,

- Do you have any errors in the browser console?
- What's your output for `jupyter labextension list`?
- Can you please list the packages installed in your virtual env?

**Comment by HuifangYeo** [comment]
Hi Thibault,

I didn't manage to check the first point. Will note them when it reoccurs.

- `jupyter labextension list`:
```
JupyterLab v3.0.0
C:\Apps\miniconda3\envs\atoti_51\share\jupyter\labextensions
        atoti-jupyterlab v0.5.1 enabled ok
        @jupyter-widgets/jupyterlab-manager v3.0.0 enabled ok (python, jupyterlab_widgets)
```

- Virtual env package list:
<details><pre>
(atoti_51) C:\Users\Yeo Hui Fang>conda list
# packages in environment at C:\Apps\miniconda3\envs\atoti_51:
#
# Name                    Version                   Build  Channel
abseil-cpp                20200923.2           h0e60522_1    conda-forge
anyio                     2.0.2            py39hcbf5309_4    conda-forge
argon2-cffi               20.1.0           py39hb82d6ee_2    conda-forge
arrow-cpp                 2.0.0           py39h59ff000_16_cpu    conda-forge
async_generator           1.10                       py_0    conda-forge
atoti                     0.5.1                   7f32d5e    https://conda.atoti.io
atoti-aws                 0.5.1                   7f32d5e    https://conda.atoti.io
atoti-jupyterlab          0.5.1                   7f32d5e    https://conda.atoti.io
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
aws-c-cal                 0.4.5                hd67dfff_4    conda-forge
aws-c-common              0.4.65               h8ffe710_0    conda-forge
aws-c-event-stream        0.2.6                hc969853_1    conda-forge
aws-c-io                  0.8.0                h9f364c9_0    conda-forge
aws-checksums             0.1.10               h9f364c9_2    conda-forge
aws-sdk-cpp               1.8.122              h0223788_0    conda-forge
babel                     2.9.0              pyhd3deb0d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
bleach                    3.2.1              pyh9f0ad1d_0    conda-forge
brotli                    1.0.9                h0e60522_4    conda-forge
brotlipy                  0.7.0           py39hb82d6ee_1001    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
c-ares                    1.17.1               h8ffe710_0    conda-forge
ca-certificates           2020.12.5            h5b45459_0    conda-forge
certifi                   2020.12.5        py39hcbf5309_1    conda-forge
cffi                      1.14.4           py39h0878f49_1    conda-forge
chardet                   4.0.0            py39hcbf5309_1    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
confuse                   1.4.0                    pypi_0    pypi
cryptography              3.3.1            py39h0878f49_1    conda-forge
cycler                    0.10.0                   pypi_0    pypi
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
gflags                    2.2.2             ha925a31_1004    conda-forge
glog                      0.4.0                h0174b99_3    conda-forge
graphviz                  2.38.0            h6538335_1011    conda-forge
grpc-cpp                  1.34.1               hd7e1925_0    conda-forge
htmlmin                   0.1.12                   pypi_0    pypi
idna                      2.10               pyh9f0ad1d_0    conda-forge
imagehash                 4.2.0                    pypi_0    pypi
imbalanced-learn          0.7.0                    pypi_0    pypi
imblearn                  0.0                      pypi_0    pypi
importlib-metadata        3.4.0            py39hcbf5309_0    conda-forge
importlib_metadata        3.4.0                hd8ed1ab_0    conda-forge
intel-openmp              2020.3             h57928b3_311    conda-forge
ipykernel                 5.4.2            py39he1a07ef_0    conda-forge
ipython                   7.19.0           py39h832f523_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.3                    pypi_0    pypi
jdk4py                    11.0.9.0                 py39_0    https://conda.atoti.io
jedi                      0.18.0           py39hcbf5309_2    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
joblib                    1.0.0                    pypi_0    pypi
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                3.2.0                      py_2    conda-forge
jupyter_client            6.1.11             pyhd8ed1ab_1    conda-forge
jupyter_core              4.7.0            py39hcbf5309_0    conda-forge
jupyter_server            1.2.1            py39hcbf5309_0    conda-forge
jupyterlab                3.0.0              pyh5fc60ef_0    conda-forge/label/jupyterlab_rc
jupyterlab-widgets        1.0.0                    pypi_0    pypi
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.0.0              pyh70d38ba_0    conda-forge/label/jupyterlab_server_rc
kiwisolver                1.3.1                    pypi_0    pypi
krb5                      1.17.2               hbae68bd_0    conda-forge
libblas                   3.9.0                     7_mkl    conda-forge
libcblas                  3.9.0                     7_mkl    conda-forge
libcurl                   7.71.1               h4b64cdc_8    conda-forge
liblapack                 3.9.0                     7_mkl    conda-forge
libprotobuf               3.14.0               h7755175_0    conda-forge
libsodium                 1.0.18               h8d14728_1    conda-forge
libssh2                   1.9.0                hb06d900_5    conda-forge
libthrift                 0.13.0               hdfef310_6    conda-forge
libutf8proc               2.6.1                hcb41399_0    conda-forge
llvmlite                  0.35.0                   pypi_0    pypi
lz4-c                     1.9.3                h8ffe710_0    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markupsafe                1.1.1            py39hb82d6ee_3    conda-forge
matplotlib                3.3.3                    pypi_0    pypi
missingno                 0.4.2                    pypi_0    pypi
mistune                   0.8.4           py39hb82d6ee_1003    conda-forge
mkl                       2020.4             hb70f87d_311    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
nbclassic                 0.2.6              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.1                      py_0    conda-forge
nbconvert                 6.0.7            py39hcbf5309_3    conda-forge
nbformat                  5.0.8                      py_0    conda-forge
nest-asyncio              1.4.3              pyhd8ed1ab_0    conda-forge
networkx                  2.5                      pypi_0    pypi
notebook                  6.2.0            py39hcbf5309_0    conda-forge
numba                     0.53.0dev0      np1.16py3.9h5685391_gddc6d35c2_465    numba/label/dev
numpy                     1.19.5           py39h6635163_1    conda-forge
openssl                   1.1.1i               h8ffe710_0    conda-forge
packaging                 20.8               pyhd3deb0d_0    conda-forge
pandas                    1.2.0            py39h2e25243_0    conda-forge
pandas-profiling          2.10.0                   pypi_0    pypi
pandoc                    2.11.3.2             h8ffe710_0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.1              pyhd8ed1ab_0    conda-forge
phik                      0.11.0                   pypi_0    pypi
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.1.0                    pypi_0    pypi
pip                       20.3.3             pyhd8ed1ab_0    conda-forge
prometheus_client         0.9.0              pyhd3deb0d_0    conda-forge
prompt-toolkit            3.0.10             pyha770c72_0    conda-forge
pyarrow                   2.0.0           py39hf9247be_16_cpu    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.7.3              pyhd8ed1ab_0    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyrsistent                0.17.3           py39hb82d6ee_2    conda-forge
pysocks                   1.7.1            py39hcbf5309_3    conda-forge
python                    3.9.1           h7840368_3_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.9                      1_cp39    conda-forge
pytz                      2020.5             pyhd8ed1ab_0    conda-forge
pywavelets                1.1.1                    pypi_0    pypi
pywin32                   228              py39hb82d6ee_1    conda-forge
pywinpty                  0.5.7            py39hde42818_1    conda-forge
pyyaml                    5.3.1            py39hb82d6ee_2    conda-forge
pyzmq                     20.0.0           py39h6e08c89_1    conda-forge
re2                       2020.11.01           h0e60522_0    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
scikit-learn              0.24.0                   pypi_0    pypi
scipy                     1.6.0                    pypi_0    pypi
seaborn                   0.11.1                   pypi_0    pypi
send2trash                1.5.0                      py_0    conda-forge
setuptools                49.6.0           py39hcbf5309_3    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sklearn                   0.0                      pypi_0    pypi
snappy                    1.1.8                ha925a31_3    conda-forge
sniffio                   1.2.0            py39hcbf5309_1    conda-forge
sqlite                    3.34.0               h8ffe710_0    conda-forge
tangled-up-in-unicode     0.0.6                    pypi_0    pypi
terminado                 0.9.2            py39hcbf5309_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
threadpoolctl             2.1.0                    pypi_0    pypi
tk                        8.6.10               h8ffe710_1    conda-forge
tornado                   6.1              py39hb82d6ee_1    conda-forge
tqdm                      4.56.0                   pypi_0    pypi
traitlets                 5.0.5                      py_0    conda-forge
typeguard                 2.9.1                      py_0    conda-forge
typing_extensions         3.7.4.3                    py_0    conda-forge
tzdata                    2020f                he74cb21_0    conda-forge
urllib3                   1.26.2             pyhd8ed1ab_0    conda-forge
vc                        14.2                 hb210afc_2    conda-forge
visions                   0.6.0                    pypi_0    pypi
vs2015_runtime            14.28.29325          h5e1d092_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
widgetsnbextension        3.5.1                    pypi_0    pypi
win_inet_pton             1.1.0            py39hcbf5309_2    conda-forge
wincertstore              0.2             py39hcbf5309_1006    conda-forge
winpty                    0.4.3                         4    conda-forge
xz                        5.2.5                h62dcd97_1    conda-forge
yaml                      0.2.5                he774522_0    conda-forge
zeromq                    4.3.3                h0e60522_3    conda-forge
zipp                      3.4.0                      py_0    conda-forge
zlib                      1.2.11            h62dcd97_1010    conda-forge
zstd                      1.4.8                h4e2f164_1    conda-forge
</pre></details>

**Comment by HuifangYeo** [comment]
Hi @tibdex,

the console threw a lot of messages:

![image](https://user-images.githubusercontent.com/35210495/104892815-93ded280-59ad-11eb-97a0-e94f95b283da.png)

![image](https://user-images.githubusercontent.com/35210495/104892865-a527df00-59ad-11eb-886a-550be552a96d.png)

**Comment by tibdex** [comment]
It looks like you're facing MDX issues. Did you write the MDX by hand or was it entirely generated from the UI?

**Comment by HuifangYeo** [comment]
Hi @tibdex,

I managed to replicate one scenario where it will be gone. I accidentally selected a chart that is not suitable for my data for instance. 
![lost_of_extension](https://user-images.githubusercontent.com/35210495/105000502-622b4180-5a69-11eb-9b23-11996d339080.gif)

As to your questions, I can't be sure as I need to manipulate the cell meta properties to add in the filters I want.

**Comment by tibdex** [comment]
The crash occurring when switching to an incompatible widget type will be fixed in the next release.
